### ISSUE TITLE
Not compatible with PyTorch version 1.7.1

### ISSUE THREAD

**Issue by kaiyikang** [speechbrain/speechbrain#972]
Hi,

Firstly, I have to say thank you for your project, that's a great job.

When I follow the [tutorial](https://colab.research.google.com/drive/1aFgzrUv3udM_gNJNUoLaHIm78QHtxdIz?usp=sharing#scrollTo=6xcAJ4OlYZCh) at local and run step 4 "Speech Recognizer", I have this issue:

`File "/anaconda3/envs/sb/lib/python3.8/site-packages/speechbrain/decoders/seq2seq.py", line 709, in forward    torch.div(candidates, vocab_size, rounding_mode="floor")                                                            
TypeError: div() got an unexpected keyword argument 'rounding_mode'`

- PyTorch version is 1.7.1.
- Python version is 3.8

When I check PyTorch 1.7.1 [document](https://pytorch.org/docs/1.7.1/generated/torch.div.html?highlight=div#torch.div) of `div()` function, it has no argument 'rounding_mode'.
So I replace `div()` with `torch.floor_divide()` in speechbrain, it works. 

Naturally, updating PyTorch to the latest version is a best solution,  but for this, I need to update my GPU and cuda environment.

**Comment by TParcollet** [comment]
Hey hi, we indeed are aware of this issue, but unsure yet on how to manage it properly ..

**Comment by TParcollet** [comment]
do you know if ``torch.floor_divide()`` is compatible with 1.8+ as well ?

**Comment by kaiyikang** [comment]
> do you know if `torch.floor_divide()` is compatible with 1.8+ as well ?

I checked the 1.8 document. Here is the [link](https://pytorch.org/docs/1.8.0/generated/torch.floor_divide.html?highlight=div#torch.floor_divide)

Yes, `torch.floor_divide()` still exists in 1.8 version. But it has warning info:

`This function’s name is a misnomer. It actually rounds the quotient towards zero instead of taking its floor. This behavior will be deprecated in a future PyTorch release.` 

And I also checked 1.9 version. it exists but the warning changed:

`torch.floor_divide() is deprecated and will be removed in a future PyTorch release. Its name is a misnomer because it actually rounds the quotient towards zero instead of taking its floor. To keep the current behavior use torch.div() with rounding_mode='trunc'. To actually perform floor division, use torch.div() with rounding_mode='floor'.`

In conclusion, `div(rounding_mode="floor")` works only in PyTorch Version >=1.8. For 1.7, we should use `torch.floor_divide()`

**Comment by TParcollet** [comment]
Ah crap, so we will have to drop 1.7 support then ...

**Comment by csukuangfj** [comment]
I think you can use an `if .. else ..` statement to choose which `div` to use depending on the version of PyTorch.

**Comment by mravanelli** [comment]
Note that according to the requirements we suggest pytorch 1.8. We actually support pytorch 1.9, which is working totally fine except for some annoying warnings triggered internally to pytorch 1.9.
